src/addons/gif.cpp

```diff
@@ -0,0 +1,117 @@
+#include "configs/base64.h"
+#include "addons/gif.h"
+#include <hardware/flash.h>
+#include "FlashPROM.h"
+#include "FlashPROM.cpp"
+#include "storagemanager.h"
+
+extern char __flash_binary_end;
+
+bool GifAddon::available() {
+    const DisplayOptions& displayOptions = Storage::getInstance().getDisplayOptions();
+    const GifOptions& gifOptions = Storage::getInstance().getAddonOptions().gifOptions;
+    return displayOptions.enabled && gifOptions.enabled;
+};
+
+void GifAddon::setup() {
+    setCurrentFrameFor(0);
+};
+
+bool GifAddon::setupForWrite(size_t numberOfBytes)
+{
+    if (get_core_num() != 0)
+        return false;
+
+    const uint32_t numberOfFrames = numberOfBytes / sizeof(SplashFrame);
+
+    const size_t framesMaxAddress = EEPROM_ADDRESS_START - XIP_BASE - GIF_EEPROM_DISTANCE;
```

Comment:
yeah doing eeprom address modification inside of an add-on is really hacky.

Reply:
No address is being modified